### PR TITLE
e2e Validation for Modeling Rule fields in Schema/TestData

### PR DESCRIPTION
**Pull request by GuyAfik**
## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-7064

## Description
1. Validate that the schema types are aligned with the TestData types.
2. Add ignore mechanism to ignore errors when testing modeling rules.

### CODE REVIEW

File: demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py [hunk 1]
```diff
@@ -266,6 +272,95 @@ def validate_expected_values(
...
+        bool: {"type": "boolean", "is_array": False},
+    }
+
+    schema_dataset_to_events = {}
+
+    # map each dataset from the schema to the correct events that has the same dataset
+    for dataset in schema.keys():
+        schema_dataset_to_events[dataset] = [
+            d.event_data for d in test_data.data if d.dataset == dataset
+        ]
```
Comment (JudahSchwartz) [inline demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py:302]: use a dict comprehension

**Review by JudahSchwartz — APPROVED** [review]
looks great! Please wait on @daryakoval's approval

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,6 @@
 # Changelog
 ## Unreleased
+* Added support for the **modeling-rules test** command to validate whether the modeling rules schema mappings are aligned with the test-data mappings.
```
Comment (JudahSchwartz) [inline CHANGELOG.md:3]:
```suggestion
* The **modeling-rules test** command will now validate that the modeling rules schema mappings are aligned with the test-data mappings.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,6 @@
 # Changelog
 ## Unreleased
+* Fixed an issue where **lint** and **validate** commands failed on integrations and scripts that use docker images that are not available in the Docker Hub but exist locally.
 * The **modeling-rules test** command will now validate that the modeling rules schema mappings are aligned with the test-data mappings.
```
Comment (daryakoval) [inline CHANGELOG.md]: Why not your addition in green?

File: demisto_sdk/commands/test_content/test_modeling_rule/tests/test_modeling_rule_test.py [hunk 4]
```diff
@@ -926,14 +926,20 @@ def test_the_test_modeling_rule_command_results_with_ignored_validations(self, p
...
 
         test_data_file = pack.modeling_rules[0].testdata
 
         fake_test_data = TestData.parse_file(path_to_fake_test_data_file.as_posix())
         test_data_file.write_as_text(fake_test_data.json(indent=4))
-        test_data_file.update({"ignored_validations": [Validations.SCHEMA_TYPES_ALIGNED_WITH_TEST_DATA.value]})
+        test_data_file.update(
+            {
+                "ignored_validations": [
+                    Validations.SCHEMA_TYPES_ALIGNED_WITH_TEST_DATA.value
```
Comment (daryakoval) [inline demisto_sdk/commands/test_content/test_modeling_rule/tests/test_modeling_rule_test.py:939]: If your class inherits from str as well, then you will able to use it without `.value`

class Validations(Enum, str):
    SCHEMA_TYPES_ALIGNED_WITH_TEST_DATA = "schema_test_data_types"

```suggestion
                    Validations.SCHEMA_TYPES_ALIGNED_WITH_TEST_DATA
```

File: demisto_sdk/commands/test_content/xsiam_tools/test_data.py [hunk 5]
```diff
@@ -1,16 +1,16 @@
+from enum import Enum
 from typing import Any, Dict, List, Optional
 from uuid import UUID, uuid4
 
 from pydantic import BaseModel, Field, validator
-from enum import Enum
 
 
 class Validations(Enum):
```
Comment (daryakoval) [inline demisto_sdk/commands/test_content/xsiam_tools/test_data.py:8]:
```suggestion
class Validations(Enum, str):
```

File: demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py [hunk 6]
```diff
@@ -619,6 +709,31 @@ def validate_modeling_rule(
...
+        if schema_path := mr_entity.schema_path:
+            with open(mr_entity.schema_path) as schema_file:
+                schema = json.load(schema_file)
+        else:
+            raise FileNotFoundError(
+                f"schema file does not exist in path {mr_entity.schema_path}"
+            )
+
+        if (
+            Validations.SCHEMA_TYPES_ALIGNED_WITH_TEST_DATA.value
```
Comment (daryakoval) [inline demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py:721]: If the class inherits from str, you don't need `.value`

```suggestion
            Validations.SCHEMA_TYPES_ALIGNED_WITH_TEST_DATA
```

File: demisto_sdk/commands/test_content/xsiam_tools/test_data.py [hunk 7]
```diff
@@ -1,9 +1,18 @@
+from enum import Enum
 from typing import Any, Dict, List, Optional
 from uuid import UUID, uuid4
 
 from pydantic import BaseModel, Field, validator
 
 
+class Validations(Enum):
```
Comment (daryakoval) [inline demisto_sdk/commands/test_content/xsiam_tools/test_data.py:8]:
```suggestion
class Validations(Enum, str):
```

File: demisto_sdk/commands/test_content/test_modeling_rule/tests/test_modeling_rule_test.py [hunk 8]
```diff
@@ -1300,3 +1511,194 @@ def test_day_suffix(day, suffix):
     )
 
     assert day_suffix(day) == suffix
+
+
+class TestValidateSchemaAlignedWithTestData:
+    @pytest.mark.parametrize(
+        "event_data, schema_file",
```
Comment (daryakoval) [inline demisto_sdk/commands/test_content/test_modeling_rule/tests/test_modeling_rule_test.py:1518]: Please add a test that catches the case where we have the same field in 2 test data events with different types.

**Review by daryakoval — COMMENTED** [review]
Great work :)

File: CHANGELOG.md (hunk 3 above)
Comment (GuyAfik) [reply CHANGELOG.md]: should be good

**Review by JudahSchwartz — APPROVED** [review]
Great work!

**Pull request merged**